lib/modules/manager/gomod/package-tree.ts

```diff
@@ -0,0 +1,187 @@
+  visitedModules.set(packageFileName, false);
+
+  for (const dependent of dependents) {
+    recursivelyGetDependentGoModFiles(dependent, graph, visitedModules);
+  }
+}
+
+/**
+ * Parse local replace directives from a go.mod file
+ */
+async function parseLocalReplaceDirectives(
+  goModFile: string,
+): Promise<{ targetGoModPath: string | null }[]> {
+  const content = await readLocalFile(goModFile, 'utf8');
+  if (!content) {
+    return [];
+  }
+
+  const directives: { targetGoModPath: string | null }[] = [];
+  const lines = content.split(newlineRegex);
+  const blockRegex = regEx(
+    /^\s*(?<module>[^\s]+)\s*=>\s*(?<replacement>[^\s]+)(?:\s+(?<version>[^\s]+))?(?:\s*\/\/.*)?$/,
+  );
+  let inReplaceBlock = false;
+  for (const line of lines) {
+    const trimmedLine = line.trim();
+
+    // Check if we're starting a multiline replace block
+    if (trimmedLine.startsWith('replace (')) {
+      inReplaceBlock = true;
+      continue;
+    }
+
+    // Check if we're ending a multiline replace block
+    if (inReplaceBlock && trimmedLine === ')') {
+      inReplaceBlock = false;
+      continue;
+    }
+
+    // Parse replace directive (either single line or inside multiline block)
+    let replaceMatches = replaceRegex.exec(line)?.groups;
+
+    // For multiline blocks, the keyword might be missing
+    if (inReplaceBlock && !replaceMatches) {
+      // Try parsing without the replace keyword
+      replaceMatches = blockRegex.exec(line)?.groups;
+    }
+
+    if (replaceMatches) {
+      const { replacement } = replaceMatches;
+      const depName = trimQuotes(replacement);
+
+      // Only process local dependencies (relative paths)
+      if (depName.startsWith('./') || depName.startsWith('../')) {
+        const targetGoModPath = reframeRelativePathToRootOfRepo(
+          goModFile,
+          depName,
+        );
+        const targetGoModFile = upath.join(targetGoModPath, 'go.mod');
+        directives.push({ targetGoModPath: targetGoModFile });
+      }
+    }
+  }
+
+  return directives;
+}
+
+/**
+ * Take the path relative from a go.mod file, and make it relative from the root of the repo
+ */
+function reframeRelativePathToRootOfRepo(
+  dependentGoModRelativePath: string,
+  replaceReference: string,
+): string {
+  const virtualRepoRoot = '/';
+  const absoluteDependentGoModPath = upath.resolve(
+    virtualRepoRoot,
+    dependentGoModRelativePath,
+  );
+  const absoluteReplaceReferencePath = upath.resolve(
+    upath.dirname(absoluteDependentGoModPath),
+    replaceReference,
+  );
+  const relativeReplaceReferencePath = upath.relative(
+    virtualRepoRoot,
+    absoluteReplaceReferencePath,
+  );
+
+  return relativeReplaceReferencePath;
+}
+
+/**
+ * Get a list of go.mod files in the repository
+ */
+async function getAllGoModFiles(): Promise<string[]> {
+  const allFiles = await scm.getFileList();
+  const filteredGoModFiles = allFiles.filter((file) => file.endsWith('go.mod'));
```

Comment:
```suggestion
  const filteredGoModFiles = allFiles.filter(minimatchFilter('go.mod', { matchBase: true, nocase: true }));
```

otherwise you find `rego.mod` and other unrelated files